src/IISIntegration/src/AspNetCoreModuleV2/CommonLib/Environment.cpp

```diff
@@ -60,3 +60,52 @@ Environment::GetEnvironmentVariableValue(const std::wstring & str)
 
     return expandedStr;
 }
+
+std::wstring Environment::GetCurrentDirectoryValue()
+{
+    DWORD requestedSize = GetCurrentDirectory(0, nullptr);
+    if (requestedSize == 0)
+    {
+        throw std::system_error(GetLastError(), std::system_category(), "GetCurrentDirectory");
+    }
+
+    std::wstring expandedStr;
+    do
+    {
+        expandedStr.resize(requestedSize);
+        requestedSize = GetCurrentDirectory(requestedSize, expandedStr.data());
+        if (requestedSize == 0)
+        {
+            throw std::system_error(GetLastError(), std::system_category(), "GetCurrentDirectory");
+        }
+    } while (expandedStr.size() != requestedSize + 1);
```

Comment:
Is the while loop present to catch cases where the value of GetCurrentDirectory changes between two calls?